lib/src/widgets/flutter_deck_progress_indicator.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_deck/src/flutter_deck.dart';
+import 'package:flutter_deck/src/flutter_deck_router.dart';
+
+/// A linear progress indicator that shows the current progress of the
+/// presentation.
+///
+/// The progress is calculated by the number of total steps in the presentation
+/// and the current step.
+class FlutterDeckProgressIndicator extends StatefulWidget {
+  /// Creates a progress indicator with a solid color.
+  const FlutterDeckProgressIndicator.solid({Color? color})
+      : this._(color: color);
+
+  /// Creates a progress indicator with a gradient.
+  const FlutterDeckProgressIndicator.gradient(Gradient gradient)
+      : this._(gradient: gradient);
+
+  /// A private constructor for the [FlutterDeckProgressIndicator] widget.
+  ///
+  /// This constructor is private because it should not be used directly.
+  /// Instead, use one of the public constructors.
+  const FlutterDeckProgressIndicator._({
+    this.color,
+    this.gradient,
+  });
+
+  /// The progress indicator color.
+  ///
+  /// This value is only provided via the [FlutterDeckProgressIndicator.solid]
+  /// constructor.
+  final Color? color;
+
+  /// The progress indicator gradient.
+  ///
+  /// This value is only provided
+  /// via the [FlutterDeckProgressIndicator.gradient] constructor.
```

Comment:
Strange comment formatting over there.

Reply:
Actually it was overflowing the length of comment i.e. 80 characters so it was giving warning

Reply:
What do you mean? If you check other comments, the break is put after the last word that fits and not somewhere random in the middle. Like:

This value is only provided via the
[FlutterDeckProgressIndicator.gradient] constructor.

Reply:
Okay gotcha